mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/work/spi/ConfiguredSearchIndexingPlanFilter.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.pojo.work.spi;
+
+import java.util.Set;
+
+import org.hibernate.search.mapper.pojo.model.spi.PojoRawTypeIdentifier;
+
+public interface ConfiguredSearchIndexingPlanFilter {
+
+	boolean isIncluded(PojoRawTypeIdentifier<?> typeIdentifier);
+
+	default boolean supportsEventQueue() {
+		return true;
+	}
+
+	static ConfiguredSearchIndexingPlanFilter create(
+			Set<PojoRawTypeIdentifier<?>> includes,
+			Set<PojoRawTypeIdentifier<?>> excludes
+	) {
+		if ( includes.isEmpty() ) {
+			return ExcludeAll.INSTANCE;
+		}
+		if ( excludes.isEmpty() ) {
+			return IncludeAll.INSTANCE;
+		}
+		return new Filter( includes );
+	}
+
+	class IncludeAll implements ConfiguredSearchIndexingPlanFilter {
+
+		public static final IncludeAll INSTANCE = new IncludeAll();
+
+		@Override
+		public boolean isIncluded(PojoRawTypeIdentifier<?> typeIdentifier) {
+			return true;
+		}
+
+		@Override
+		public boolean supportsEventQueue() {
```

Comment:
`supportsAsyncProcessing` would better reflect the intent, I think.

Reply:
👍🏻 makes sense 😃. I wanted it to be aligned with `ConfiguredAutomaticIndexingStrategy#usesEventQueue()`

Reply:
Ok, will do  😃 👍🏻